File: template-repo/templates/nextjs-mysql/lib/ai/AiConfig.ts
import 'server-only'

function readNumber(value: string | undefined, fallback: number) {
  const parsed = Number(value)
  return Number.isFinite(parsed) ? parsed : fallback
}

export class AiConfig {
  baseUrl = process.env.AI_BASE_URL ?? 'https://dashscope.aliyuncs.com/compatible-mode/v1'
  apiKey = process.env.AI_API_KEY ?? ''
  model = process.env.AI_MODEL ?? 'qwen-plus'
  embeddingModel = process.env.AI_EMBEDDING_MODEL ?? 'text-embedding-v3'
  timeoutSeconds = readNumber(process.env.AI_TIMEOUT_SECONDS, 60)
  maxTokens = readNumber(process.env.AI_MAX_TOKENS, 4096)
  temperature = readNumber(process.env.AI_TEMPERATURE, 0.7)
}

export const aiConfig = new AiConfig()

File: template-repo/templates/springboot-vue3-mysql/backend/src/main/java/com/smartark/template/ai/AiConfig.java
package com.smartark.template.ai;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

/**
 * AI configuration properties.
 * Set via environment variables or application.yml under "ai.*".
 */
@Component
@ConfigurationProperties(prefix = "ai")
public class AiConfig {

    private String baseUrl = "https://dashscope.aliyuncs.com/compatible-mode/v1";
    private String apiKey = "";
    private String model = "qwen-plus";
    private String embeddingModel = "text-embedding-v3";
    private int timeoutSeconds = 60;
    private int maxTokens = 4096;
    private double temperature = 0.7;

    public String getBaseUrl() { return baseUrl; }
    public void setBaseUrl(String baseUrl) { this.baseUrl = baseUrl; }

    public String getApiKey() { return apiKey; }
    public void setApiKey(String apiKey) { this.apiKey = apiKey; }

    public String getModel() { return model; }
    public void setModel(String model) { this.model = model; }

    public String getEmbeddingModel() { return embeddingModel; }
    public void setEmbeddingModel(String embeddingModel) { this.embeddingModel = embeddingModel; }

    public int getTimeoutSeconds() { return timeoutSeconds; }
    public void setTimeoutSeconds(int timeoutSeconds) { this.timeoutSeconds = timeoutSeconds; }

    public int getMaxTokens() { return maxTokens; }
    public void setMaxTokens(int maxTokens) { this.maxTokens = maxTokens; }

    public double getTemperature() { return temperature; }
    public void setTemperature(double temperature) { this.temperature = temperature; }
}
